vm/src/builtins/type.rs

```diff
@@ -594,6 +596,21 @@ impl PyType {
         let typ = Self::new_verbose_ref(name.as_str(), base, bases, attributes, slots, metatype)
             .map_err(|e| vm.new_type_error(e))?;
 
+        typ.attributes
+            .write()
+            .get(identifier!(vm, __classcell__))
+            .map(|cell| {
+                let cell = PyCellRef::try_from_object(vm, cell.clone()).map_err(|_| {
+                    vm.new_type_error(format!(
+                        "__classcell__ must be a nonlocal cell, not {}",
+                        cell.class().name()
+                    ))
+                })?;
+                cell.set(Some(typ.clone().to_pyobject(vm)));
```

Comment:
This line has side-effect and the closure doesn't return anything. Using if-let statement would be better than map